specification/ml/_types/Job.ts

```diff
@@ -49,28 +50,33 @@ export class JobStatistics {
 }
 
 export class Job {
```

Comment:
This change is making everything optional, should wherever this type is used instead use `JobConfig`?

Reply:
Hmm... it could indeed extend `JobConfig` (with the only difference being that `analysis_config` is required in `JobConfig`). Is this what you mean?

I ended up making all fields optional after a number of validation iterations where they were all successively reported as "missing required field"

Reply:
I'm wondering if this issue you're running into is the same as what I was trying to fix in https://github.com/elastic/elasticsearch-specification/pull/1283 but couldn't complete due to needing to use an `OverloadOf` class as a type of one of the subproperties IIRC?

For now can we maintain two classes `Job` and `JobRead` where `JobRead` has all properties as they are today and another class `Job` with the definition you have in this PR? What are your thoughts on this approach?